sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -264,6 +269,18 @@ public EventHubClientBuilder clientOptions(ClientOptions clientOptions) {
         return this;
     }
 
+    /**
+     * Sets the client identifier.
+     *
+     * @param identifier Identifier for the client.
+     * @return The updated {@link EventHubClientBuilder} object.
+     */
+    @Override
+    public EventHubClientBuilder identifier(String identifier) {
+        this.identifier = identifier;
```

Comment:
Other limitation aside, there at least need a `Objects.requireNonNull`?

Reply:
About the length limit, Jesse's answer is no, the reason is that the service is responsible for any validation of the identifier.

Sure, I will add `Objects.requireNonNull` and use `{@inheritDoc}`.